classes/class-truelayer-config-editor.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Config Editor Class file.
+ *
+ * @package TrueLayer_For_WooCommerce/Classes
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * The TrueLayer Config Editor class
+ */
+class TrueLayer_Config_Editor {
+	private $config_path;
+	private $backup_path;
+
+	/**
+	 * Class constructor to setup the class.
+	 *
+	 * @param string $path The path to the file you want to edit. Defaults to wp-config.php.
+	 * @param string $backup_path The path to the backup file. Defaults to wp-config-backup.php.
+	 */
+	public function __construct( $path = '', $backup_path = '' ) {
+		$this->config_path = empty( $path ) ? ABSPATH . 'wp-config.php' : $path;
+		$this->backup_path = empty( $backup_path ) ? ABSPATH . 'wp-config-backup.php' : $backup_path;
+	}
+
+	/**
+	 * Add a specific key with a value to the wp-config.php file as a definition.
+	 *
+	 * @param string $key_name The name of the key to add.
+	 * @param string $key_value The value of the key to add.
+	 *
+	 * @return bool True if the key was added, false if it was not.
+	 */
+	public function add_key( $key_name, $key_value ) {
+		$key_name  = $this->sanitize_key_name( $key_name );
+		$key_value = $this->sanitize_key_value( $key_value );
+
+		// Create a backup for the config file first.
+		$backup_result = $this->create_backup();
+
+		// If the backup failed, then we cannot continue, since it might mean the write protection of the folder is on.
+		if ( ! $backup_result ) {
+			return false; // Backup failed
+		}
+
+		$config_data = file( $this->config_path, FILE_IGNORE_NEW_LINES );
+
+		// Check if key already exists
+		foreach ( $config_data as $line ) {
+			if ( strpos( $line, $key_name ) !== false ) {
+				return false; // Key already exists
+			}
+		}
+
+		// Find the place to add the new key
+		$insert_at = 0;
+		foreach ( $config_data as $index => $line ) {
+
+			// Try to insert the new key after the nonce salt definition
+			if ( strpos( $line, 'NONCE_SALT' ) !== false ) {
+				$insert_at = $index + 1;
+				break;
+			}
+
+			// If we did not find the nonce salt definition, then test against the "stop editing" comment
+			if ( strpos( $line, '/* That\'s all, stop editing! Happy publishing. */' ) !== false ) {
+				$insert_at = $index;
+				break;
+			}
+		}
+
+		// Insert the new key
+		array_splice( $config_data, $insert_at, 0, "define( '$key_name', '$key_value' );" );
```

Comment:
Just a note here if we do end up allowing non alpha numerical characters in the key value, we should guard againt `'` or escape it.

Reply:
Yes agreed, based on the answer to the question before ill make the change needed for this.

Reply:
Solved by stripping the `'` character from the value, as i said before, if you prefer i escape it instead, just let me know.